File: lpc3131/examples/misc/adc/adc.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>
#include <time.h>

unsigned char buffer[256];

void select_gpa1()
{
   /* select ADC 1 */
   fd = open("/dev/lpc313x_adc", O_RDWR);
   write(fd, "1", 2);  /* "1\0" */
   close(fd);
}

int get_adc()
{
   fd = open("/dev/lpc313x_adc", O_RDONLY);
   n = read(fd, buffer, 256);
   close(fd);
   return n;
}

File: lpc3131/examples/misc/adc/adc.py
import os

DEVICE = '/dev/lpc313x_adc'

def select_gpa1():
   fd = os.open(DEVICE, os.O_RDWR)
   os.write(fd, "0x0001")
   os.close(fd)

def get_adc():
   fd = os.open(DEVICE, os.O_RDONLY)
   av = os.read(fd, 256)
   os.close(fd)
   return av[:-1]  # strip off trailing '\n'

if __name__ == "__main__":
   select_gpa1()
   ad_val = get_adc()
   print ad_val



